Mysql installation script erroneously defines `ofMessageArchive`.`isPMforJID` as `NOT NULL`

`ofMessageArchive`.`isPMforJID`'s content is expected to be `NULL` when an archived message is _not_ a private message. The installation script for MySQL however, defines the column as `NOT NULL`, causing exceptions like the one below to be logged when message archiving is enabled.

The column was added in version 2.2.0.

Interestingly enough, the update script (number 7) does _not_ have this problem. This causes the issue to occur only on new installations (as opposed to those that upgraded from a previous version).

```
2021.02.12 10:51:58 ERROR [archive-service-worker-3]: org.jivesoftware.openfire.archive.ConversationManager - Unable to archive message data!
java.sql.BatchUpdateException: Column 'isPMforJID' cannot be null
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at com.mysql.cj.util.Util.handleNewInstance(Util.java:192) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.util.Util.getInstance(Util.java:167) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.util.Util.getInstance(Util.java:174) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.exceptions.SQLError.createBatchUpdateException(SQLError.java:224) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeBatchedInserts(ClientPreparedStatement.java:755) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeBatchInternal(ClientPreparedStatement.java:426) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.StatementImpl.executeBatch(StatementImpl.java:796) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at org.apache.commons.dbcp2.DelegatingStatement.executeBatch(DelegatingStatement.java:223) ~[commons-dbcp2-2.6.0.jar:2.6.0]
        at org.apache.commons.dbcp2.DelegatingStatement.executeBatch(DelegatingStatement.java:223) ~[commons-dbcp2-2.6.0.jar:2.6.0]
        at org.jivesoftware.openfire.archive.ConversationManager$MessageArchivingRunnable.store(ConversationManager.java:1199) [monitoring-2.2.0.jar:?]
        at org.jivesoftware.openfire.archive.Archiver.run(Archiver.java:154) [xmppserver-4.6.2.jar:4.6.2]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.sql.SQLIntegrityConstraintViolationException: Column 'isPMforJID' cannot be null
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        at com.mysql.cj.jdbc.ClientPreparedStatement.executeBatchedInserts(ClientPreparedStatement.java:716) ~[mysql-connector-java-8.0.21.jar:8.0.21]
        ... 11 more
```

--- a/changelog.html
+++ b/changelog.html
@@ -52,6 +52,7 @@ Monitoring Plugin Changelog
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/165'>Issue #165</a>] - Fix ordering of MAM results</li>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/167'>Issue #167</a>] - Deal with empty stanzas in db when processing MAM queries</li>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/174'>Issue #174</a>] - Upgrade jackson libraryto 2.12.1</li>
+    <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/175'>Issue #175</a>] - Fix database installation script for MySQL (bug introduced in 2.2.0 that affected new installations)</li>
 </ul>
 
 <p><b>2.2.0</b> -- January 6, 2021</p>

--- a/src/database/monitoring_db2.sql
+++ b/src/database/monitoring_db2.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        INTEGER      NOT NULL,

--- a/src/database/monitoring_hsqldb.sql
+++ b/src/database/monitoring_hsqldb.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        BIGINT        NOT NULL,

--- a/src/database/monitoring_mysql.sql
+++ b/src/database/monitoring_mysql.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        BIGINT        NOT NULL,
@@ -35,7 +35,7 @@ CREATE TABLE ofMessageArchive (
    sentDate          BIGINT           NOT NULL,
    stanza			 TEXT			  NULL,
    body              TEXT             NULL,
-   isPMforJID        VARCHAR(255)     NOT NULL,
+   isPMforJID        VARCHAR(255)     NULL,
    INDEX ofMessageArchive_con_idx (conversationID),
    INDEX ofMessageArchive_fromjid_idx (fromJID),
    INDEX ofMessageArchive_tojid_idx (toJID),

--- a/src/database/monitoring_oracle.sql
+++ b/src/database/monitoring_oracle.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        INTEGER        NOT NULL,

--- a/src/database/monitoring_postgresql.sql
+++ b/src/database/monitoring_postgresql.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        INTEGER       NOT NULL,

--- a/src/database/monitoring_sqlserver.sql
+++ b/src/database/monitoring_sqlserver.sql
@@ -1,5 +1,5 @@
 
-INSERT INTO ofVersion (name, version) VALUES ('monitoring', 7);
+INSERT INTO ofVersion (name, version) VALUES ('monitoring', 8);
 
 CREATE TABLE ofConversation (
   conversationID        BIGINT         NOT NULL,

--- a/src/database/upgrade/8/monitoring_db2.sql
+++ b/src/database/upgrade/8/monitoring_db2.sql
@@ -1,0 +1,4 @@
+-- This is a MySQL-specific update.
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

--- a/src/database/upgrade/8/monitoring_hsqldb.sql
+++ b/src/database/upgrade/8/monitoring_hsqldb.sql
@@ -1,0 +1,4 @@
+-- This is a MySQL-specific update.
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

--- a/src/database/upgrade/8/monitoring_mysql.sql
+++ b/src/database/upgrade/8/monitoring_mysql.sql
@@ -1,0 +1,4 @@
+ALTER TABLE ofMessageArchive MODIFY COLUMN isPMforJID VARCHAR(255) NULL;
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

--- a/src/database/upgrade/8/monitoring_oracle.sql
+++ b/src/database/upgrade/8/monitoring_oracle.sql
@@ -1,0 +1,4 @@
+-- This is a MySQL-specific update.
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

--- a/src/database/upgrade/8/monitoring_postgresql.sql
+++ b/src/database/upgrade/8/monitoring_postgresql.sql
@@ -1,0 +1,4 @@
+-- This is a MySQL-specific update.
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

--- a/src/database/upgrade/8/monitoring_sqlserver.sql
+++ b/src/database/upgrade/8/monitoring_sqlserver.sql
@@ -1,0 +1,4 @@
+-- This is a MySQL-specific update.
+
+-- Update database version
+UPDATE ofVersion SET version = 8 WHERE name = 'monitoring';

